src/Table2Beta/Table.tsx

```diff
@@ -508,8 +508,19 @@ export class Table2Beta extends React.Component<Props, State> {
     }
 
     const { displayedData, numPages, allRows } = this._getDisplayedData();
-    selectedRows = new Set(selectedRows && allRows);
-    this.setState({ selectedRows });
+
+    // This occurs when a new external filter is applied.
+    // We want selectedRows to change if the filter affects the selection
+    const newSelectedRows = new Set(selectedRows);
+    newSelectedRows.forEach((item) => {
+      if (!allRows.includes(item)) {
+        newSelectedRows.delete(item);
+      }
+    });
+    if (newSelectedRows.size !== selectedRows.size) {
+      this.setState({ selectedRows: newSelectedRows });
```

Comment:
Is there a chance the size is the same but the sets are different?

Reply:
Nope! Line `514` declares newSelectedRows as a copy of selectedRows, so they have the same data. From there, newSelectedRows can only get smaller.